File: scripts/quarter-fetcher/anime.go
package main

import (
	"time"
)

type Program struct {
	Time string
	Name string
}

type Link struct {
	Title string
	URL   string
}

type Song struct {
	Title   string
	Details map[string]string
}

type OpenGraphImage struct {
	URL string
}

type OpenGraph struct {
	Images      []*OpenGraphImage
	Description string
}

type Episode struct {
	Channel  string
	StartAt  time.Time
	Duration time.Duration
	Number   int
	Subtitle string
}

type Timetable []Episode

type Anime struct {
	Id         string
	Title      string
	Website    string
	OpenGraph  *OpenGraph
	Staff      map[string]string
	Opening    *Song
	Ending     *Song
	Cast       map[string]string
	Programs   []*Program
	Links      []*Link
	Channels   []string
	Timetables []Timetable
}

File: src/models/anime.ts
interface IAnimeSong {
   Title: string;
   Details: { [key: string]: string };
}

interface IProgram {
   Time: string;
   Name: string;
}

interface ILink {
   Title: string;
   URL: string;
}

interface ITimetable {
   Channel: string;
   StartAt: string;
   Duration: number;
   Number: number;
   Subtitle: string;
}

interface IAnime {
   Id: string;
   Title: string;
   Website: string;
   OpenGraph?: any;
   Staff: { [key: string]: string };
   Opening?: IAnimeSong;
   Ending?: IAnimeSong;
   Cast: { [key: string]: string };
   Programs: IProgram[];
   Links: ILink[];
   Channels: string[];
   Timetables: ITimetable[];
}

export type { IAnime, ITimetable };